tortoise/tests/test_filtering.py

```diff
@@ -0,0 +1,49 @@
+from tortoise.contrib.test import TestCase
+from tortoise.query_utils import Q
+from tortoise.tests.testmodels import Event, Team, Tournament
+
+
+class TestFiltering(TestCase):
+    async def test_filtering(self):
+        tournament = Tournament(name='Tournament')
+        await tournament.save()
+
+        second_tournament = Tournament(name='Tournament 2')
+        await second_tournament.save()
+
+        event_first = Event(name='1', tournament=tournament)
+        await event_first.save()
+        event_second = Event(name='2', tournament=second_tournament)
+        await event_second.save()
+        event_third = Event(name='3', tournament=tournament)
+        await event_third.save()
+        event_forth = Event(name='4', tournament=second_tournament)
+        await event_forth.save()
+
+        team_first = Team(name='First')
+        await team_first.save()
+        team_second = Team(name='Second')
+        await team_second.save()
+
+        await team_first.events.add(event_first)
+        await event_second.participants.add(team_second)
+
+        found_events = await Event.filter(
+            Q(id__in=[event_first.id, event_second.id])
+            | Q(name='3')
+        ).filter(participants__not=team_second.id).order_by('tournament__id').distinct()
```

Comment:
Here is another example of postgres `for SELECT DISTINCT, ORDER BY expressions must appear in select list` - but sadly, I don't think we can do much about it here, even though postgres could guess that tournament__id is unique and it isn't ambiguous it still doesn't want to work that way.
I guess we should change ordering here to just `tournament_id`